aim/api/status.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) 2016 Cisco Systems
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+
+class Status(object):
+
+    def is_build(self):
+        """Returns True if the current status is transient."""
+        raise
+
+    def is_error(self):
+        """Returns True if current status is an error state."""
+        raise
+
+
+class AciStatus(Status):
+    """Status of an AIM resource that is mapped to ACI object.
+
+    Following attributes are available:
+    * create_status - Indicates whether ACI object was created or not
```

Comment:
Is this intended to reflect whether the object has been created in the APIC? Does it also reflect whether the APIC has been updated with any changes made to the object after it was created? I was expecting something called sync_status that reflected both - basically indicating whether all pending changes to the resource have been synced to the APIC. Would it make sense to rename/redefine this to sync_status?